packages/dev/core/src/Rendering/iblCdfGenerator.ts

```diff
@@ -213,29 +233,52 @@ export class IblCdfGenerator {
         this._cdfyPT.autoClear = false;
         this._cdfyPT.setTexture("iblSource", this._iblSource as Texture);
         this._cdfyPT.setInt("iblHeight", size.height);
-        if (this._iblSource.isCube) {
-            this._cdfyPT.defines = "#define IBL_USE_CUBE_MAP\n";
-        }
+        this._cdfyPT.wrapV = Constants.TEXTURE_CLAMP_ADDRESSMODE;
         this._cdfyPT.refreshRate = 0;
         this._icdfyPT = new ProceduralTexture("icdfyTexture", { width: size.width, height: size.height }, "iblIcdfy", this._scene, icdfOptions, false, false);
         this._icdfyPT.autoClear = false;
         this._icdfyPT.setTexture("cdfy", this._cdfyPT);
         this._icdfyPT.refreshRate = 0;
+        this._icdfyPT.wrapV = Constants.TEXTURE_CLAMP_ADDRESSMODE;
         this._cdfxPT = new ProceduralTexture("cdfxTexture", { width: size.width + 1, height: 1 }, "iblCdfx", this._scene, cdfOptions, false, false);
         this._cdfxPT.autoClear = false;
         this._cdfxPT.setTexture("cdfy", this._cdfyPT);
         this._cdfxPT.refreshRate = 0;
+        this._cdfxPT.wrapU = Constants.TEXTURE_CLAMP_ADDRESSMODE;
         this._icdfxPT = new ProceduralTexture("icdfxTexture", { width: size.width, height: 1 }, "iblIcdfx", this._scene, icdfOptions, false, false);
         this._icdfxPT.autoClear = false;
         this._icdfxPT.setTexture("cdfx", this._cdfxPT);
         this._icdfxPT.refreshRate = 0;
+        this._icdfxPT.wrapU = Constants.TEXTURE_CLAMP_ADDRESSMODE;
+
+        this._normalizationPT = new ProceduralTexture("normalizationTexture", { width: 1, height: 1 }, "iblNormalization", this._scene, cdfOptions, false, false);
+        this._normalizationPT.autoClear = false;
+        this._normalizationPT.setTexture("iblSource", this._iblSource as Texture);

```

Comment:
Instead of this cast (multiple occurrences in the file), you should update `ProceduralTexture` so that `ProceduralTexture.setTexture` takes a `ThinTexture`. You will also have to change the type of `ProceduralTexture._textures` to `{ [key: string]: ThinTexture }`.